### DISCUSSION TITLE
Megathread: Detect dark theme

### DISCUSSION THREAD

**Discussion by Gusted** [darkreader/darkreader#9016] [General]
Please report websites whereby Dark Reader failed to detect a dark theme.

To enable this detection:

- Click Dark Reader icon.
- Click Dev tools (in the bottom-right corner).
- Click Preview new design.
- Then go to Settings, click Detect dark theme.

Please note that this isn't stable(hence this thread).

**Comment by Gusted** [comment]
https://slightknack.github.io/rust-abi-wiki/intro/intro.html originally reported in #9003

**Comment by Gusted** [comment]
https://www.gimp.org/downloads/devel/ originally reported in #9009

**Comment by Milky063** [comment]
So I had Darkread for some time, but this latest update it doen't work on https://www.msn.com/ on firefox. not sure if this is a known bug.

**Reply by Gusted** [reply]
Is this related to the detection of dark theme?

**Comment by Gusted** [comment]
https://fishshell.com/docs/current/cmds/test.html Originally reported in #8991

**Comment by Milky063** [comment]
yes....it doesn't work on msn.com but every other site it works perfectly fine.
________________________________

**Comment by Elaborendum** [comment]
https://en.wikipedia.org/wiki/Main_Page?useskin=timeless Sometimes doesn't work with Wikipedia's Timeless skin specifically, haven't tested with the other skins.
https://wiki.greasespot.net/ Originally reported in #8472
https://inkscape.org/forums/ Originally reported in #8472
https://mail.tutanota.com/mail Tutanota Mail when logged in
https://www.murphysmultiverse.com/

**Comment by Gusted** [comment]
https://authors.curseforge.com/dashboard Reported in #8980

**Comment by osfanbuff63** [comment]
https://osfanbuff63.github.io or any other site using the [minimal-mistakes-jekyll](https://github.com/mmistakes/minimal-mistakes) `dark` skin

**Comment by osfanbuff63** [comment]
This is the opposite (as in, detected a dark theme when there was none) and it worked after a refresh: https://www.thetrevorproject.org

Edit: Also is a bit buggy on that site when turning it off and on

**Comment by Elaborendum** [comment]
https://www.android-x86.org/ and basically all its subpages.

**Comment by Gusted** [comment]
https://next-episode.net/ Reported in #9307

**Comment by Elaborendum** [comment]
https://gravatar.com/ Only the main site

**Comment by JosephChataignon** [comment]
https://josephchataignon.github.io/ (my test website), I used `:root { color-scheme: light dark; }` to signal that I natively support a dark theme

**Comment by Gusted** [comment]
https://github.com/darkreader/darkreader/issues/9469 php.net/manual

**Comment by Gusted** [comment]
Reported in https://github.com/darkreader/darkreader/issues/9550, https://majorgeeks.com

**Comment by darkmode111** [comment]
https://arstechnica.com/ Detects it as dark theme

**Comment by FadeMind** [comment]
https://cezarywalenciuk.pl/blog/programing

![20220824-1661370418](https://user-images.githubusercontent.com/9846948/186509585-5a8da197-8fa2-4bf0-b2c0-0d5188bf6181.png)

**Comment by FadeMind** [comment]
https://store.epicgames.com/

Dark theme is not detected at all

**Comment by Gusted** [comment]
https://deepnote.com/ from #9826

**Comment by Gamrix** [comment]
https://outlook.live.com/ 
Dark theme is not detected when the website's dark mode is toggled, even after refresh.

**Comment by Gusted** [comment]
https://www.arangodb.com/docs reported in #10211

**Comment by luca-demma** [comment]
https://www.hdblog.it/ doesn't work

**Comment by auroraanna** [comment]
Dark theme not detected on
- https://annaaurora.eu/

**Comment by Gusted** [comment]
https://mail.proton.me/u/0/inbox Reported in #10322

**Comment by gshpychka** [comment]
Dark theme not detected on https://console.aws.amazon.com (have to be logged in)

**Comment by antwan** [comment]
Same for https://www.linkedin.com and many many others...

I think the original approach to detect if a dark mode is applied by using [color detection](https://github.com/darkreader/darkreader/blob/main/src/inject/detector.ts#L11) on the `<html>` and `<body>` elements is somewhat flaky. It sometimes requires a delay until document is effectively painted, and relies on elements that are not necessary the most visible ones, creating numerous false positives and negatives, and sometimes random flickering.

Instead it would probably be more reliable to detect whether those main elements [CSS rules](https://gist.github.com/ssafejava/6605832) are subject to some `prefer-color-scheme` [media queries](https://stackoverflow.com/questions/15696124/accessing-css-media-query-rules-via-javascript-dom), or have specific class/attributes keywords such as `.dark-theme`, `body[data-dark-theme]`, etc., and only then fallback onto the color heuristic when nothing is detected.

This color detection heuristic could also easily be improved, using random / grid polling points in the visible screen to assess an average of the most visible elements [real background color](https://stackoverflow.com/questions/12576084/getting-the-real-background-color-of-an-element), instead of assuming the `<body>` and `<html>` are only what's on screen.
```js
// Poll points in a 10×10 grid over visible layout
let visibleColors = [];
[...Array(10).keys()].map(i => Math.floor((i + 0.5)/10  * window.innerWidth)).forEach(x => {
    [...Array(10).keys()].map(i => Math.floor((i + 0.5)/10  * window.innerHeight)).forEach(y => {
        visibleColors.push(realBackgroundColor(document.elementFromPoint(x, y)));
    });
});

// Return average brightness
return visibleColors.reduce((p, v) => p + 0.01 * getSRGBLightness(parseColorWithCache(v)), 0)
```

**Reply by alexanderby** [reply]
Nice idea with `document.elementFromPoint(x, y)`, I will try experimenting with it.

**Comment by lica98** [comment]
https://www.flashscore.com/

**Reply by lica98** [reply]
also on https://www.soccer24.com/ it detects the dark theme but the site needs to be reloaded to work

**Comment by imsodin** [comment]
I guess false positives are ok too  - the majority of times that it didn't work it was a false positive. Example is this:
https://scrumguides.org/scrum-guide.html
I presume it happens due to the dark background of the bar at the top? At least that seems like a common theme for false positives, that some larger area, but not the main area, has a dark background.

**Comment by SlyAceZeta** [comment]
Dark theme incorrectly detected at https://psnprofiles.com/ including player profiles, game trophy lists, and the forums.

**Comment by cobrabr** [comment]
Dark theme not detected on https://calendar.cron.com/.

**Comment by cobrabr** [comment]
Dark theme not detected on https://meet.google.com/ — this is a tricky one, because while the main page is (correctly) detected as not being in dark mode, the actual meeting page _is_ in dark mode (at least for the most part).

**Comment by cobrabr** [comment]
Dark theme not detected on https://app.thestorygraph.com/.

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://www.themoviedb.org/.

**Comment by cobrabr** [comment]
Dark theme not detected on https://learn.standardresume.co/.

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://sevendistrictscoffee.com/.

**Comment by cobrabr** [comment]
Dark theme not detected on https://riverside.fm/.

**Comment by gregsdennis** [comment]
Literally any Github.com site.  Dark theme is enabled through GH, but DarkReader still modifies it.
StackOverflow.com.  Same.

**Reply by gshpychka** [reply]
> Literally any Github.com site.  Dark theme is enabled through GH, but DarkReader still modifies it.
> 
> StackOverflow.com.  Same.

Same here with GitHub, but then it works after a refresh.

**Reply by cobrabr** [reply]
For me, GitHub works fine on Chrome but not on Edge. Refreshing sometimes fixes it, but it eventually goes back to not working again.

**Comment by cobrabr** [comment]
Dark theme not detected on https://discord.com/, at least not on the login page.

**Comment by cobrabr** [comment]
Dark theme not detected on https://docs.piral.io/ and https://docs.piral.cloud/. The upside is that they both have a quick toggle for light mode and Dark Reader's conversion of that to dark mode looks better than their "native" dark mode. 😆

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://azure.microsoft.com/.

**Comment by cobrabr** [comment]
Dark theme not detected on https://app.roll20.net/editor/.

**Comment by cobrabr** [comment]
Dark theme not detected on https://www.microsoft365.com/.

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://www.switchbacktravel.com/.

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://www.activestate.com/products/python/.

**Comment by cobrabr** [comment]
Dark theme not detected on https://outlook.office.com/.

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://knightedgemedia.com/.

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://www.livenation.com/.

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://www.topresume.com/.

**Comment by cobrabr** [comment]
Dark theme incorrectly detected on https://www.imdb.com/.

**Comment by cobrabr** [comment]
Dark theme not detected on https://regex101.com/.

**Comment by ngoclong19** [comment]
Dark Reader incorrectly detect as dark theme on
`https://www.steamgifts.com/`

**Comment by YoshiRulz** [comment]
https://bleepingcomputer.com

**Comment by ghost** [comment]
https://github.com

**Comment by ghost** [comment]
https://bard.google.com

**Comment by ghost** [comment]
All Mastodon instances, including:

- [mastodon.online](https://mastodon.online)
- [mastodon.social](https://mastodon.social)
- [mas.to](https://mas.to)
- [mindly.social](https://mindly.social)
- [toot.community](https://toot.community)
- [masto.ai](https://masto.ai)
- [hachyderm.io](https://hachyderm.io)
- [mstdn.ca](https://mstdn.ca)
- [mastodon.scot](https://mastodon.scot)
- [mastodon.nl](https://mastodon.nl)
- [fosstodon.org](https://fosstodon.org)
- [mstdn.jp](https://mstdn.jp)
- [mastodon.world](https://mastodon.world)
- [ohai.social](https://ohai.social)
- [glitch.lgbt](https://glitch.lgbt)
- [mastodonapp.uk](https://mastodonapp.uk)

[and many more here.](https://joinmastodon.org/servers)

**Comment by Backspaze** [comment]
*.proton.me for Proton's different subdomains for all their services such as [mail.proton.me](https://mail.proton.me/), [calendar.proton.me](https://calendar.proton.me/) and so on.

Randomly on first load or reload of the page DR doesn't detect the dark theme and instead enables it's own dark theme. But if you then click the URL once in DR, you can see how the theme changes to Proton's dark theme, but DR still says **Enabled for current website** under the URL. If you click the URL in DR once more or click outside DR and then click on DR again, the text under the URL will change to **Dark theme detected on page**.

This is tested on Windows 11 Pro 22H2 in Firefox 115.0.2 (64-bit) and DR 4.9.64.

**Comment by gsalyer** [comment]
Incorrectly detected on [fitgirl-repacks.site](https://fitgirl-repacks.site/)

**Comment by daxpedda** [comment]
Not detected on https://reddit.com when using the "Dark Mode" option available when logged in.

**Comment by daxpedda** [comment]
Incorrectly detected dark theme on https://gamedev.net.
Interestingly enough the page is very very light, the darkest thing on it are light blue buttons.

**Comment by daxpedda** [comment]
Doesn't detect dark theme on https://www.cybenetics.com.

**Comment by noizo** [comment]
Incorrectly detected on https://someproject.atlassian.net/jira.
Overwrites build in dark mode on https://someproject.atlassian.net/jira/, and if i change brightness or contrast settings, it immediately detects dark theme. Disabling it is breaking other services, where there is no dark mode, like https://someproject.atlassian.net/wiki/.

**Comment by ghost** [comment]
https://app.slack.com

**Comment by ghost** [comment]
https://reduced.to

**Comment by ghost** [comment]
https://raindrop.io
https://app.raindrop.io

**Comment by ghost** [comment]
https://web.telegram.org

**Comment by ghost** [comment]
https://facebook.com

**Comment by ghost** [comment]
https://www.radboudumc.nl/patientenzorg

Dark Reader thinks the site has a dark theme, but in fact the site only has a light mode.

**Comment by jtbrough** [comment]
smartsheet.com

Dark Reader reports that a dark mode is available but that isn't true.

**Comment by ghost** [comment]
https://www.alarabiya.net

**Comment by ghost** [comment]
https://regex101.com
Dark theme not detected

**Comment by daxpedda** [comment]
https://www.arroyo.dev
Falsely detects dark theme.

**Reply by ghost** [reply]
Also, some text is invisible when dark mode is enabled.

**Comment by NamelessGO** [comment]
https://flathub.org/
Not detected
The website colors is synced via light/dark theme of the browser

**Comment by ghost** [comment]
https://www.cia.gov
Falsely detects dark mode

**Comment by ghost** [comment]
https://www.dictionary.com/e/word-of-the-day
Falsely detects dark mode

**Comment by ghost** [comment]
https://search.brave.com
Does not detect auto dark mode.

**Comment by ghost** [comment]
https://www.epa.gov
Falsely detects dark mode

**Comment by ghost** [comment]
https://anonymfile.com
Does not detect dark mode.

**Comment by ghost** [comment]
https://www.gotquestions.org
Falsely detects dark mode

**Comment by ghost** [comment]
Seems like Dark Reader falsely detects dark mode on some Blogspot sites.

Some examples:
https://philosophicaldisquisitions.blogspot.com
https://nomoremister.blogspot.com

**Comment by lica98** [comment]
[sporcle.com](https://www.sporcle.com/)

**Comment by ghost** [comment]
https://thefederalist.com
Falsely detects dark mode.

**Comment by ghost** [comment]
https://www.lingoblocks.com
 Falsely detects dark mode.

**Comment by ghost** [comment]
https://www.theverge.com
 Does not detect dark mode.

**Comment by ghost** [comment]
https://messages.google.com
 Does not detect dark mode.

**Comment by alexanderby** [comment]
Thank you guys for sharing the links, I'm looking into the issue now.

**Comment by ghost** [comment]
https://www.si.edu
Falsely detects dark theme.

**Comment by alexanderby** [comment]
Thanks everyone again. There are 2 (3 actually) reasons why the false detection is happening (or not happening):
1. Some websites have black background on their body and root elements, and then the content has white background.
2. Some websites enable their own dark theme after Dark Reader's detection has already passed.
3. There's a bug in Chrome itself: when Dark Reader's styles are synchronously disabled they still have effect for some milliseconds in some conditions that I haven't figured out yet.
4. For performance reasons we run the detection when there is at least a single style or when the root or body elements have colours in `style` attribute.

For that reason I've added [Detector Hints config](https://github.com/darkreader/darkreader/blob/main/src/config/detector-hints.config). For most of the websites it should be easy to fix the detection:
1. `TARGET` means a selector of an element that has some attribute that defines dark theme existence on the web page (usually it's the root `html` element).
2. `MATCH` is a selector that matches the dark theme (for some websites it is `dark-mode` class or `[data-dark-mode]` attribute or similar).
3. `NO DARK THEME` means the website has no dark theme and the default detection is false positive.

I will be filling this list, let's see if it will be useful. It also should slightly improve the performance for websites that work correctly, so I added a few popular sites too.

**Comment by lica98** [comment]
https://www.vecernji.hr/

**Comment by lumynou5** [comment]
https://twitter.com/
https://web.telegram.org/
Cannot detect if the theme is set to dark by user

**Comment by lumynou5** [comment]
![screenshot](https://github.com/darkreader/darkreader/assets/84374077/7109a53a-f720-44bd-a501-ed86de8f4d97)
https://developer.chrome.com/docs/
Falsely detect (work after disable and then enable)

**Comment by lumynou5** [comment]
https://www.twitch.tv/ It seems to make the whole video player black.

**Comment by alexanderby** [comment]
Thank you guys for the websites. I went through all of them and added a manual fix for most of them, an update will be published soon. I see some common things and will try making the automatic detection better.

**Comment by net** [comment]
https://github.com

Doesn't detect GitHub's dark theme when active via **sync with system**.

<img width="953" alt="Screenshot 2023-12-15 at 17 22 12" src="https://github.com/darkreader/darkreader/assets/6983821/73aa7756-2cd4-4c81-862f-bfd02d2f972a">

**Reply by lumynou5** [reply]
#11995

**Comment by lumynou5** [comment]
https://crowdin.com/ not detected.

**Comment by lumynou5** [comment]
https://meiert.com/ not detected and make page light.

**Comment by gguman** [comment]
Is it possible to invert picture-in-picture video?

**Comment by alexanderby** [comment]
@lumynou5 Thanks, I made some tweaks for those websites and will release them soon.

**Comment by alexanderby** [comment]
@gguman I'm afraid the videos like these exist outside of web page environment, so it is not possible.

**Reply by gshpychka** [reply]
Hey, just a suggestion - replying directly to a message instead of creating a new one might help with readability later.

**Comment by ghost** [comment]
https://sojo.net
 Falsely detects dark theme.

**Comment by johann-backup** [comment]
Dark theme not detected on canva.com . Have been advised the following:

> We expose this on the <body> through a simple CSS class (theme dark) so something along the lines of "Don't filter if body.theme.dark".

**Comment by ghost** [comment]
https://support.mozilla.org
Site stays white despite Dark Reader turned on.

**Reply by ngoclong19** [reply]
If you are using Firefox, that site is a restricted domain, and by default, add-ons will be disable.
See below link for more information and how to re-enable the add-on for these domains.
https://github.com/gorhill/uBlock/wiki/Privileged-Pages

**Reply by gshpychka** [reply]
@ngoclong19 the specific link you provided refers to a separate feature - Mozilla domains are not quarantined, and the protection there cannot be disabled.

**Reply by ngoclong19** [reply]
@gshpychka, Thank you for pointing out my mistake! I've updated my reply.

It looks like you can disable the protection in Firefox, domain wide, not add-on specific (please be sure about the risks).
Only the protection on the extensions store page and settings pages cannot be disable.
`https://github.com/darkreader/darkreader/blob/da592e1b91854ee011428e463cbc3cf3781934f6/src/background/utils/extension-api.ts#L5`

**Comment by lumynou5** [comment]
https://developer.mozilla.org/

**Reply by ghost** [reply]
Interesting, it properly detects it for me. Perhaps you've added it as an exception?

![Screenshot 2024-01-29 at 1 45 35 AM](https://github.com/darkreader/darkreader/assets/52135169/5c9c89ba-ea1e-48b0-99e1-1b4e5a9b423a)

**Reply by lumynou5** [reply]
It seems not being detected sometimes.

![image](https://github.com/darkreader/darkreader/assets/84374077/5ee370cf-3433-48fb-b80b-4ea4c5008592)
![image](https://github.com/darkreader/darkreader/assets/84374077/92145601-4de7-4498-ab98-f41399eae27e)

**Comment by ghost** [comment]
https://www.freetaxusa.com/taxes2023
Falsely detects dark theme.

![Screenshot 2024-01-31 at 15-44-57 FreeTaxUSA® -- Prior Year AGI](https://github.com/darkreader/darkreader/assets/52135169/4ff037ad-de19-43f7-9fd1-9e050d228aaa)

**Comment by RobHelgeson** [comment]
https://hibernate.org

If I toggle off dark mode detection, it looks great.

If it matters, I am on Safari.

**Comment by Opa-** [comment]
Some protonmail services are not correctly detected

**Not** detected:
🔴 https://mail.proton.me
🔴 https://drive.proton.me
🔴 https://pass.proton.me

Dark theme detected :
🟢 https://calendar.proton.me
🟢 https://account.proton.me/u/1/vpn/dashboard

**Comment by 0xifarouk** [comment]
❌ `https://{your-organization}.atlassian.net/jira/` is not being detected.

**Comment by ghost** [comment]
https://us.startpage.com is not being detected.

**Comment by ghost** [comment]
https://send.vis.ee is not being detected.

**Comment by pvinis** [comment]
https://md.quad.codes  is not being detected.

**Comment by quarkquartet** [comment]
Twitter not detected...

**Reply by ghost** [reply]
Probably because of the migration to x.com.

**Comment by quarkquartet** [comment]
https://pkmer.cn/ is not being detected.

Furthermore recently I found dark reader is enabled for google (native dark enabled) when I first open it (i.e. not detecting google dark mode), and if I refresh then it goes back to normal. Not sure whether it's a problem from my side or not.

**Reply by lumynou5** [reply]
Check what the pannel shows. If it says a dark theme is detected, but the dark reader is still enabled, then it's a known issue #11990, which seems almost always happening on newly opened tabs.

**Comment by Johnnycyan** [comment]
Sorry if this is the wrong place to ask but as a website dev what can I do to tell Dark Reader my website is dark themed?

**Reply by ngoclong19** [reply]
Please see:
`https://github.com/darkreader/darkreader/blob/main/CONTRIBUTING.md#disabling-dark-reader-on-your-site`

**Comment by count0x30** [comment]
https://www.startpage.com is not being detected when Theme is set to "System default".  So, when my Linux/Mac/Win system is in dark mode, Startpage correctly uses its dark theme, but this isn't detected by Dark Reader.

However, if the Startpage theme is explicitly set to "Dark", then Dark Reader detects it.

**Comment by starsep** [comment]
https://demo.jellyfin.org/stable/web

Dark theme detection doesn't work for Jellyfin: usually self-hosted, above public instance url.
This causes issues: https://github.com/jellyfin/jellyfin-web/issues/3542#issuecomment-1375238940

**Comment by Awesome-0** [comment]
https://duckduckgo.com/ is also not being detected. Cheers!

**Reply by gshpychka** [reply]
Fixed for me.

**Comment by gshpychka** [comment]
Dark theme not detected: https://tmuxcheatsheet.com/

**Comment by Diwannist** [comment]
https://sdfl-umber.vercel.app/

**Comment by gshpychka** [comment]
https://opentelemetry.io/

**Reply by gshpychka** [reply]
This is fixed

**Comment by adamhotep** [comment]
I was pretty frustrated with Dark Reader ignoring my `:root { color-scheme: dark light; }` (even as the first item in the only inline style!) or `<html style="color-scheme: dark light">`. I didn't want to force it with `<meta name="darkreader-lock">` as that's inelegant and won't work for other dark-mode extensions. I eventually found this to be reliable:

```html
<meta name="color-scheme" content="light dark" />
```

I hope this helps other web developers.

**Reply by lumynou5** [reply]
I think this is not the intention of the megathread, you may open a new discussion for it.

BTW, your first thought probably can be implemented like `getComputedStyle(document.documentElement)['color-scheme'].includes('dark')` as a part of the dark theme detector, so if the page can be rendered in dark color scheme, DarkReader won't activate.  The pro is the standard feature allowing the detector to work on more websites than the meta tag.  However, the current detector based on background color seems fine, so I wonder why you need to tell DarkReader manually rather than letting it detect the background color automatically.

**Reply by adamhotep** [reply]
The webpage in question has no colors at all, instead relying on the browser defaults that are keyed by the color scheme ... so there's nothing to detect if Dark Reader is ignoring the stated color scheme. Employing `getComputedStyle(document.body)` may solve that, though it'll be fooled by sites that have a dark body as a border around a light `<div>` that contains all of the content.

**Comment by Bilge** [comment]
https://steam250.com

In the "old design" it does not detect dark mode. In the "new design", it seems that it does, but still makes (bad) changes to the design anyway. It should not make any changes at all.

**Reply by Myshor** [reply]
Seems always dark. I will add this URL to Dark Sites. 😉

**Reply by Myshor** [reply]
Added entry in https://github.com/darkreader/darkreader/pull/14153

**Comment by raingros** [comment]
Incorrect detection of a dark theme on
   
https://thesweetbits.com/tools/imazing-review/  

after the page was initially displayed in dark mode with Dark Reader. The page switches to a light theme after I enter a few words in the address bar. Then Dark Reader recognizes a non-existens dark theme. I experience the same issue with several other websites.

**Reply by Myshor** [reply]
1. Can you make some screen capture video and share it here?
2. What browser you use because I can not reproduce your described problem on Firefox.

**Reply by raingros** [reply]
Re 1. Screen capture video: https://github.com/user-attachments/assets/f0f10f94-0c52-4fc3-b767-5a33652f7a93

Re 2. I use Chrome Version 139.0.7258.139 on Windows 11.

**Reply by Myshor** [reply]
Still can not reproduce but I am adding this site as NO DARK in detections configuration.

https://github.com/darkreader/darkreader/pull/14608

**Comment by grape-day** [comment]
Dark theme isn't automatically detected on fandom.com.
For example: https://postal.fandom.com/wiki/POSTAL

**Reply by Myshor** [reply]
Thanks for info. Added detector to detector-hints.config in https://github.com/darkreader/darkreader/pull/14694

**Comment by pwall2222** [comment]
https://effect.website/docs

**Reply by Myshor** [reply]
Added detections in https://github.com/darkreader/darkreader/pull/14818

**Comment by pwall2222** [comment]
All the https://proton.me suite, (the main page is light themed though).

**Reply by Myshor** [reply]
I will leave this untouched as I do not have any valid account to work. Maybe someone will have some already and could check what can be done with it.

**Reply by pwall2222** [reply]
For {`account`, `mail`, `calendar`, `drive`, `wallet`}`.proton.me` their identifier is `[data-theme-mode="dark"]` on the `html` element.
The rest of services are a bit wonky with theming (they use css classes on body, or a style element with an id, etc).

**Comment by Henonnnn** [comment]
Dark theme isn't automatically detected on Kagi ([https://kagi.com](https://kagi.com))

**Reply by Myshor** [reply]
I can add detection when Dark Theme is manually set in setting. System preference can be dark or light so I can not include this in detections.
<img width="418" height="47" alt="obraz" src="https://github.com/user-attachments/assets/0903deee-a401-4f11-a98e-227f6f0f6b94" />

**Reply by Myshor** [reply]
Added detection in https://github.com/darkreader/darkreader/pull/14819

**Comment by TheJanox824** [comment]
Dark theme is not being detected on Brave Search (https://search.brave.com) and Google Gemini (https://gemini.google.com).
Both pages are set to use the system theme, so dark theme is on by default, but Dark Reader overwrites it, making them look wrong.

**Reply by Myshor** [reply]
As we have policy to not include detection for system setting as it can be light or dark depending on user preferences in browser/system.  `search.brave.com` adds `.light` or `.dark` class in `html` tag when light or dark theme is set in brave's search options. With this it's easy to detect dark and that detection is working as expected. Setting mode option to System in Brave's search removes all classes from `html` and bases on system preferences in side's css. That's why it's not detected by Dark Reader and can not be included in dark-theme detectors.

`gemini.google.com` on the other hand is acting totally different. It sets `.dark-theme` in `body` tag and it does that in both scenarios - when dark or system is selected. Of course if user uses light theme in system/browser it sets `.light-theme`. That makes it easy to include in manual detection fixes.

Summarizing: I will add fix for `gemini.google.com` but not for `search.brave.com`.

**Reply by Myshor** [reply]
Fix for gemini added in https://github.com/darkreader/darkreader/pull/14950

**Reply by kirelagin** [reply]
@Myshor Question though: for websites like search.brave.com or kagi.com right above, why not add them to the detector hints with `SYSTEM THEME`?

**Reply by Myshor** [reply]
@kirelagin There was a PR introducing `SYSTEM THEME` related detections (which I commented on, and as far as I remember, it was never merged). But even so, it still requires matching a specific element or attribute in the HTML to confirm that the dark theme is currently active. If a site relies strictly on CSS media queries (like `prefers-color-scheme`) without altering the DOM, I still cannot use it in `detection-hints`.

Unless you know a workaround I'm missing – is there a way to force a check in `detection-hints` based purely on the browser's theme state to confirm the site is in dark mode?

**Reply by lumynou5** [reply]
I think we can just treat system theme as detected because whether the system setting is light or dark, it's what the user wants.

**Reply by kirelagin** [reply]
@Myshor What exactly is your concern? I think that, if a website is known to follow the system theme, we should just add it to the list and assume that it does so from now on. If it stops doing this (unlikely?), we just wait for someone to complain and remove it. So, I don’t see why we would need to confirm this by matching a specific element.

@lumynou5 I am not 100% sure what the reasoning behind the policy is, but, I think, some users have system theme set to light, and still want dark reader to darken the websites. Again, I don’t know the details, but the policy explicitly says that a web-site can be added to the “do not engage” list only if it is unconditionally dark.

My understanding was that `SYSTEM THEME` addresses exactly this problem because it disables Dark Reader not unconditionally, but rather only if the user has their system set to dark: https://github.com/darkreader/darkreader/blob/6b3760f6307c3fd70915013c340805ba94f141ce/src/inject/detector.ts#L157-L160.

**Reply by Myshor** [reply]
Ok... you just showed me something I didn't even know we could use, as no one has used it yet in `detection-hints`.

I tested `SYSTEM THEME` and...
It's great for sites that don't have their own manual theme toggle and switch automatically based strictly on system preferences.

But why can't we use it when a site has a manual theme switcher (even if one of the options is "System")?

Because this TAG acts globally, just like `NO DARK THEME`.
If a user manually forces a light theme on the website (or set it a long time ago), they will end up with a blinding white page, while Dark Reader will just say **"Dark theme detected on page"** and refuse to work.

I checked search.brave.com and it has a manual theme switch in settings:
<img width="1228" height="815" alt="obraz" src="https://github.com/user-attachments/assets/a4ca59c5-410e-4515-9909-2205b628f946" />

Besides, the forced dark mode for this site is already properly handled in`detection-hints` using DOM matching:
<img width="419" height="344" alt="obraz" src="https://github.com/user-attachments/assets/c2b2a400-8381-41a3-a12b-aae9a0cf2992" />

**Reply by kirelagin** [reply]
> as no one has used it yet in `detection-hints`.

Let’s fix this: https://github.com/darkreader/darkreader/pull/15396 🙂. (There use to be one website added by the original implementor of `SYSTEM THEME`, but then they removed it a year later (I think because the automatic detection started working on that website).)

So, I think, we agree that, for a website without any toggle, this is the right thing to do. Should I open a PR adding this to the contributing guide? Or is there some RFC/discussion process to go through first?

For websites with a toggle, even if it includes the “system” option, I would argue we should add them to the “always dark” list. Because, in those cases, the user has control on the website directly, as @lumynou5 was saying.

**Reply by Myshor** [reply]
> Let’s fix this: #15396 🙂. 

Not so fast. 😋 I won't close PR yet but check comment there.

**Comment by olipra** [comment]
Dark theme is false-positively detected on https://altisplay.fr/ (and possibly other landing pages made using https://taap.bio/)

**Reply by Myshor** [reply]
Added fix in https://github.com/darkreader/darkreader/pull/15787 but for altisplay.fr only.

Checked taap.bio and some od it's subpages and couldn't confirm bad dark theme detection so without exact examples I won't include it to exclude detection.

**Reply by olipra** [reply]
I found https://taap.bio/totoche and it works as expected, so it’s only an altisplay.fr issue. Thanks!

**Comment by okixk** [comment]
[qwant.com](qwant.com) has a native dark mode, but Dark Reader doesn’t detect it with “Detect dark theme” enabled. Tested on Brave and Firefox.

**Reply by Myshor** [reply]
Fix added https://github.com/darkreader/darkreader/pull/16019

- If you want to have this fix faster you can set **Synchronizee sites fixes** option to check in Advanced section of Settings.
![obraz](https://github.com/user-attachments/assets/9dea8b66-b77a-46ab-8aa6-e50d995718c3)
- With that enabled newest sitefixes will be updated every browser start. 😉
- To open all settings you must click on:
(Old/classic design) More -> All Settings
![obraz](https://github.com/user-attachments/assets/8c17b1f1-e85e-4672-8a88-85930e178e02)
(New/Preview design) Settings in popup
![obraz](https://github.com/user-attachments/assets/e4f586f8-8ce4-47bc-90d7-7451eb030439)
